src/ServiceControlInstaller.Engine/Instances/ServiceControlInstance.cs

```diff
@@ -344,8 +344,17 @@ public bool TryStopService()
                 return true;
             }
 
-            Service.Stop();
+            try
+            {
 
+                // Will throw with "The service cannot accept control messages at this time" message if the service is in a pending state 
+                Service.Stop();
+            }
+            catch (InvalidOperationException)
```

Comment:
No it should be safe.  "Stop" only requests the stop operation. It does not wait for the service to stop or report a failure if the stop operation fails.  This exception will occur if the Windows ServiceControl manager rejects the "Stop" request and that should only occur if it thinks the service it's in a pending or bad state.